proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,100 @@
+# ***Add `nuget list` Functionality to DotNet***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add `nuget list` functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+This is a customer requested functionality as discussed in this issue: https://github.com/NuGet/Home/issues/6060. Adding the functionality will aid our customers to be able to use the command when writing scripts.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+Imagine you have your dotnet CLI open. You would like to get the list of all packages from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search --source <MySource>`. It will provide you with the list of all the packages in the source `<MySource>`.
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options 
+
+| Option | Function |
+|---------|:----------|
+| `--Source` | A list of packages to search |
+| `--ExactMatch` | Return exact matches only as a search result |
+| `--Verbose` | Displays a detailed ist of information for each package |
+| `--Prerelease` | Allow prerelease packages to be shown. |
+| `--IncludeDelisted` | Allow unlisted packages to be shown |
+| `--NonInteractive` | Do not prompt for user input or confirmations.|
+| `--Help` | Show command help and usage information |
+| `--Verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+| `--ConfigFile` | The NuGet configuration file. If not specified, the hierarchy of configuration files from the current directory will be used. |
+|||
+#### **Algorithm**
+1. Read and parse arguments from user
+2. Use the nuget.exe Search API to look for the specific term in the provided source. The search filter will be composed of the `--Source`, `Search term`, `--Prerelease`, `--IncludeDelisted` values.
```

Comment:
I don't know what "the nuget.exe Search API" is.

I believe that `nuget.exe list` uses a NuGet Server V2 API (not documented), which means that it doesn't work with V3 servers, such as https://api.nuget.org/v3/index.json.  I don't think there's value in introducing a new command for a protocol we consider deprecated.

If you mean the [V3 protocol search resource](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource), then it might be worthwhile for this spec to acknowledge that search APIs don't gauruntee "correct" enumeration of all packages. Due to complexities in how search indexes work, plus a search index might be updated in the middle of a "list", some packages might get skipped, or some pacakges must get listed in multiple responses.